addon-test-support/setup-context.js

```diff
@@ -0,0 +1,40 @@
+import { run } from '@ember/runloop';
+import { set, setProperties, get, getProperties } from '@ember/object';
+import buildOwner from './build-owner';
+
+/*
+ * Responsible for:
+ *
+ * * create an owner object and set it on the provided context (e.g. this.owner)
+ * * setup this.set, this.setProperties, this.get, and this.getProperties to the provided context
+ */
+export default function(context, options = {}) {
+  let resolver = options.resolver;
+  let owner = buildOwner(resolver);
+
+  context.owner = owner;
+
+  context.set = function(key, value) {
+    let ret = run(function() {
+      return set(context, key, value);
+    });
+
+    return ret;
+  };
+
+  context.setProperties = function(hash) {
+    let ret = run(function() {
+      return setProperties(context, hash);
+    });
+
+    return ret;
+  };
+
+  context.get = function(key) {
+    return get(context, key);
+  };
+
+  context.getProperties = function(...args) {
+    return getProperties(context, args);
+  };
```

Comment:
should we also support `toggleProperty()`?

Reply:
We _could_, but it wasn't included in the initial RFC (or the now "legacy" system) so I'd like to hold off for now.  I'm not saying it requires an RFC, just that I want to implement the current thing without having a moving target...